### PR TITLE
Add scroll effect to CTA button

### PR DESCRIPTION
**Pull request by RonaldDijkstra**
Adds a scroll effect and focus on the first field, after clicking the CTA. 

(CAPP Agile Learning Page)

### CODE REVIEW

File: source/javascripts/all.js [hunk 1]
```diff
@@ -4,6 +4,8 @@
 //= require fitvids
 
 //= require _raf-polyfill
+//= require _scrollTo.min
```
Comment (sn3p) [inline source/javascripts/all.js]: I think we should try to exclude external libs from the repo:

```
source "https://rails-assets.org" do
  gem "rails-assets-jquery.scrollTo"
end
```

File: source/javascripts/all.js [hunk 2]
```diff
@@ -4,6 +4,8 @@
...
+//= require _scrollToForm
```
Comment (sn3p) [inline source/javascripts/all.js]: And dasherize our partials ;)

File: source/localizable/capp-agile-learning.nl.erb [hunk 3]
```diff
@@ -10,8 +10,7 @@ unique_for_locale: true
 
 <%= partial "capp-agile/capp-agile-testimonials" %>
 
-<section class="demo-request gray">
-  <a name="demo"></a>
+<section class="demo-request gray" id="form">
```
Comment (sn3p) [inline source/localizable/capp-agile-learning.nl.erb]: Prefer a more specific name, e.g. `id="demo-request"`.

File: source/localizable/_hero_capp-agile.erb [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
       <h1><%= t("capp-agile.heading") %></h1>
 
-      <a class="button white-green" href="#demo" data-event="demo_request"><%= t("capp-agile.action") %></a>
+      <a class="button white-green" id="cta_button" href="#form" data-event="demo_request"><%= t("capp-agile.action") %></a>
```
Comment (sn3p) [inline source/localizable/_hero_capp-agile.erb]: html/css is dasherized `id="cta-button"`

File: source/javascripts/_scrollToForm.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+jQuery(function () {
+  $('#cta_button').click(function () {
+    $.scrollTo(this.hash, 600, {
+      onAfter: function () {
+        $('#name').focus();
+      },
+    });
+    $(this.hash).find('#form').text(this.title);
```
Comment (sn3p) [inline source/javascripts/_scrollToForm.js]: Wait.. what is this sorcery?

1. This basically says `$('#form').find('#form')`, which will never be found.
2. And if this selector was found, the form content would be replaced with and empty string (the button doesn't have a title).

Also in my opinion `#name` ID is too generic. Also we could probably target the input by name:

```js
$(function () {
  'use strict';

  $('#cta_button').click(function () {
    var $target = $(this.hash);

    $.scrollTo($target, 600, {
      onAfter: function () {
        $target.find('input[name="name"]').focus();
      }
    });

    return false;
  });
});
```

File: source/javascripts/all.js (hunk 1 above)
Comment (RonaldDijkstra) [inline source/javascripts/all.js]: Dope

File: source/javascripts/all.js (hunk 2 above)
Comment (RonaldDijkstra) [inline source/javascripts/all.js]: Ok

File: source/localizable/capp-agile-learning.nl.erb (hunk 3 above)
Comment (RonaldDijkstra) [inline source/localizable/capp-agile-learning.nl.erb]: Done

File: source/localizable/_hero_capp-agile.erb (hunk 4 above)
Comment (RonaldDijkstra) [inline source/localizable/_hero_capp-agile.erb]: Ok :)

File: source/javascripts/_scrollToForm.js (hunk 5 above)
Comment (RonaldDijkstra) [inline source/javascripts/_scrollToForm.js]: Updated

**Comment by RonaldDijkstra** [comment]
Change All The Things. 

Ready again

**Pull request merged**